tests/unit/models/test_constraints.py

```diff
@@ -207,4 +214,102 @@ def test_invalid_version_str(version):
         _VersionStrModel(version=str(version))
 
 
+# endregion
+# region SpdxLicenseStr tests
+class _SpdxLicenseStrModel(pydantic.BaseModel):
+    license: SpdxLicenseStr
+
+
+@pytest.mark.parametrize(
+    ("license_str", "license_value"),
+    [
+        ("MIT", "MIT"),
+        ("GPL-3.0", "GPL-3.0-only"),
+        ("GPL-3.0+", "GPL-3.0-or-later"),
```

Comment:
hmm I'm not sure we should be converting things like `GPL-3.0` into `GPL-3.0-only`. Possibly there are already some expectations on this field from things like the snap store
Also [the spdx site](https://spdx.org/licenses/) lists `GPL-3.0+` as a valid ID, so as a user if I put in GPL-3.0+ I would expect to get GPL-3.0+ back out

Reply:
``GPL-3.0` / `GPL-3.0+` is in the deprecated section, that's probably a reason why they're transformed into a "current" valid identifier. I'm open for discussion on that, we may also return the same string and issue a warning to the user that this license is deprecated by SPDX, and they should use `XYZ` instead.

For the store tho, I don't think that is an issue, as they should respect the current (v3) list of identifiers.

Reply:
Removed conversion in https://github.com/canonical/craft-application/pull/402/commits/6f3e022f266286c77123c178814764707337fa28